### ISSUE TITLE
Rainbow agent implementation?

### ISSUE THREAD

**Issue by wszjzhang** [tensorflow/agents#42]
Along the way to enhance DQN, looks like DDQN is also getting implemented. How about the Rainbow algorithm?

**Comment by oars** [comment]
I think we have most of the pieces for rainbow, but we still need to run baselines on it and get code ready for OSS, no promises on when this will come out.

**Comment by sguada** [comment]
Hopefully we will release CategoricalDQN (C51) in a couple of weeks and then the other pieces of Rainbow.

**Comment by jionglearningzhang** [comment]
Are there any pieces missing? I can help.

**Comment by adityauser** [comment]
@sguada I would like to contribute to CategoricalDQN (C51). Moreover, I had already implemented this in my [repo](https://github.com/adityauser/Quest-RL/tree/master/Distributional-RL). You can review it also.

**Comment by nealwu** [comment]
Hi all, in case you missed it we currently have Categorical DQN (C51) available in our repo along with n-step updates: https://github.com/tensorflow/agents/tree/master/tf_agents/agents/categorical_dqn

I also implemented prioritized replay, often considered the third key component of the Rainbow agent. However we actually found that our agent performed just as well and trained more stably without prioritized replay, so we decided to exclude it from our Rainbow implementation.

Our combined C51 + n-step update agent performs just as well as other Rainbow agents on many of the Atari environments we've tested, so I'm happy to say we can close this issue.
